src/lib/snack-bar/snack-bar.ts

```diff
@@ -72,41 +74,21 @@ export class MatSnackBar {
    * @param component Component to be instantiated.
    * @param config Extra configuration for the snack bar.
    */
-  openFromComponent<T>(component: ComponentType<T>, config?: MatSnackBarConfig): MatSnackBarRef<T> {
-    const _config = {...this._defaultConfig, ...config};
-    const snackBarRef = this._attach(component, _config);
-
-    // When the snackbar is dismissed, clear the reference to it.
-    snackBarRef.afterDismissed().subscribe(() => {
-      // Clear the snackbar ref if it hasn't already been replaced by a newer snackbar.
-      if (this._openedSnackBarRef == snackBarRef) {
-        this._openedSnackBarRef = null;
-      }
-    });
-
-    if (this._openedSnackBarRef) {
-      // If a snack bar is already in view, dismiss it and enter the
-      // new snack bar after exit animation is complete.
-      this._openedSnackBarRef.afterDismissed().subscribe(() => {
-        snackBarRef.containerInstance.enter();
-      });
-      this._openedSnackBarRef.dismiss();
-    } else {
-      // If no snack bar is in view, enter the new snack bar.
-      snackBarRef.containerInstance.enter();
-    }
-
-    // If a dismiss timeout is provided, set up dismiss based on after the snackbar is opened.
-    if (_config.duration && _config.duration > 0) {
-      snackBarRef.afterOpened().subscribe(() => snackBarRef._dismissAfter(_config!.duration!));
-    }
-
-    if (_config.announcementMessage) {
-      this._live.announce(_config.announcementMessage, _config.politeness);
-    }
+  openFromComponent<T>(component: ComponentType<T>, config?: MatSnackBarConfig):
+    MatSnackBarRef<T> {
+    return this._attach(component, config) as MatSnackBarRef<T>;
+  }
 
-    this._openedSnackBarRef = snackBarRef;
-    return this._openedSnackBarRef;
+  /**
+   * Creates and dispatches a snack bar with a custom template for the content, removing any
+   * currently opened snack bars.
+   *
+   * @param template Template to be instantiated.
+   * @param config Extra configuration for the snack bar.
+   */
+  openFromTemplate(template: TemplateRef<any>, config?: MatSnackBarConfig):
```

Comment:
Point for discussion: currently the font size and color styling are attached to the `SimpleSnackBar`, which means that consumers that open through `openFromComponent` and `openFromTemplate` won't get the styling. Should we switch to attaching them on the `MatSnackBarContainer` instead?

Reply:
I think that we should not do it at all, especially if we plan on having it be different between templates and components. 

 It was done this way as the `SimpleSnackBar` is the snack bar that matches the spec and we give users full control with opening a custom snackbar.